components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/recorder/DefaultUserDeleteEventRecorder.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.user.mgt.recorder;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Default user delete event recorder is used to record the delete events in to a file in CSV format.
+ */
+public class DefaultUserDeleteEventRecorder implements UserDeleteEventRecorder {
+
+    private static final Log log = LogFactory.getLog(DefaultUserDeleteEventRecorder.class);
+
+    private static final String PATH_PROPERTY_NAME = "path";
+    private static final String COMMA = ",";
+
+    @Override
+    public void recordUserDeleteEvent(String username, String domainName, int tenantId, Date timeStamp,
+                                      Map<String, String> properties) throws RecorderException {
+
+        String path = properties.get(PATH_PROPERTY_NAME);
+
+        if (StringUtils.isEmpty(path)) {
+            throw new RecorderException("Cannot find the file path to write the CSV file.");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("CSV file path is set to: " + path);
+        }
+
+        path = IdentityUtil.fillURLPlaceholders(path);
+
+        if (log.isDebugEnabled()) {
+            log.debug("CSV path after resolving the carbon placeholders: " + path);
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(username)
+                .append(COMMA)
+                .append(domainName)
+                .append(COMMA)
+                .append(tenantId)
+                .append(COMMA)
+                .append(timeStamp.toString())
+                .append(COMMA);
+
+        // Write the given properties after above values.
+        for (Map.Entry<String, String> values : properties.entrySet()) {
+
+            // This property is specific to this recorder. No use of writing it.
+            if (values.getKey().equals(PATH_PROPERTY_NAME)) {
+                continue;
+            }
+            stringBuilder.append(values.getValue()).append(COMMA);
+        }
+
+        // Create the file if it does not exist. Open with write permission and append to the end.
+        try (OutputStream outputStream = Files.newOutputStream(Paths.get(path), StandardOpenOption.CREATE,
```

Comment:
Can we use custom logger for this, instead of handling files. This will allow someone to do the config in single location, and we get all log4j power.